test/lit/passes/gsi-debug.wast

```diff
@@ -38,9 +38,9 @@
   ;; CHECK-NEXT:  (drop
   ;; CHECK-NEXT:   ;;@ struct.c:20:2
```

Comment:
Why did the `select` have debug info even before this PR (and not the children)?

Reply:
The `select` received it because `replaceCurrent` copies debug info, and we replace the `struct.get` with the `select`. But it doesn't copy recursively into children (we've considered that before, but it's not clear if it's correct, and would be slower).

Reply:
Yes, but not recursively, just on the replacement:

https://github.com/WebAssembly/binaryen/blob/cdf8139a441c27c16eff02ccee65c463500fc00f/src/ir/debuginfo.h#L43

So the select gets it, but not the select's children.